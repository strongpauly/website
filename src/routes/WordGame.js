import React, { Component } from 'react';
import HiFrame from '../components/HiFrame';

export default class Home extends Component {

  render() {
    return <div id="wordGameContainer">
      <div className="container">
        A simple word game implemented using react.  Select words of the length suggested by the box below the grid.
        <ul>
          <li>The + will start a new game.</li>
          <li>The ♺ will reset the game.</li>
          <li>The ? will provide a hint. </li>
        </ul>
        If the game can no longer be completed, the ♺ will rotate.
        It&apos;s source can be found <a href="https://www.github.com/strongpauly/wordgame">here</a>.
      </div>
      <HiFrame src="/wordgame" title="wordgame"></HiFrame>
    </div>;
  }

}
