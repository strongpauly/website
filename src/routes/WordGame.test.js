import React from 'react';
import WordGame from './WordGame';
import {shallow} from 'enzyme';

/* eslint-env jest */

describe('WordGame', () => {
  it('renders without crashing', () => {
    const app = shallow(<WordGame/>);
    expect(app).toMatchSnapshot();
  });

  it('creates a #wordGameContainer', () => {
    const app = shallow(<WordGame/>);
    expect(app.find('#wordGameContainer')).toHaveLength(1);
  });
});
