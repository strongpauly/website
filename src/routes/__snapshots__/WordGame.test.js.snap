// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordGame renders without crashing 1`] = `
<div
  id="wordGameContainer"
>
  <div
    className="container"
  >
    A simple word game implemented using react.  Select words of the length suggested by the box below the grid.
    <ul>
      <li>
        The + will start a new game.
      </li>
      <li>
        The ♺ will reset the game.
      </li>
      <li>
        The ? will provide a hint. 
      </li>
    </ul>
    If the game can no longer be completed, the ♺ will rotate. It's source can be found 
    <a
      href="https://www.github.com/strongpauly/wordgame"
    >
      here
    </a>
    .
  </div>
  <HiFrame
    src="/wordgame"
    title="wordgame"
  />
</div>
`;
